src/zcash/zecPlugin.ts

```diff
@@ -86,12 +89,31 @@ export class ZcashTools implements EdgeCurrencyTools {
     if (typeof spendKey !== 'string') throw new Error('Invalid spendKey type')
 
     // Get current network height for the birthday height
-    const birthdayHeight = await this.getNewWalletBirthdayBlockheight()
+    const currentNetworkHeight = await this.getNewWalletBirthdayBlockheight()
+
+    let height = currentNetworkHeight
+
+    const { birthdayHeight } = opts ?? {}
+    if (birthdayHeight != null) {
+      if (
+        typeof birthdayHeight !== 'string' ||
+        !/^[1-9]\d*$/gm.test(birthdayHeight) // must be a valid blockheight
```

Comment:
Did ChatGPT come up with this regexp? I ask because it matches an array of number strings, one per line. For instance, this would be valid input for the regexp:

```
12
45
23
```

If you delete the `gm` on the end of the regexp, then it will match only a single number like "123". It's such a weirdly specific thing to add, but still buggy, that it makes me think AI was involved.

We already have an `asIntegerString` cleaner that does this validation, so you could use that + `asMaybe` if you wanted.

Reply:
Haha no I didn't use chatgpt. I grabbed it from stackoverflow because I'm still living in 2022. Will make the switch to asIntegerString